topic-controller/src/main/java/io/strimzi/controller/topic/Controller.java

```diff
@@ -616,6 +616,29 @@ public void handle(Future<Void> fut) {
         inFlight.enqueue(topicName, resultHandler, futureHandler);
     }
 
+    void onTopicPartitionsChanged(TopicName topicName, Handler<AsyncResult<Void>> resultHandler) {
+        Handler<Future<Void>> futureHandler = new Reconciliation("onTopicPartitionsChanged") {
+            @Override
+            public void handle(Future<Void> fut) {
+
+                // in order to get updated information about the partitions change we have to wait
+                // a little bit before using the Admin Client API for getting such an information
+                vertx.setTimer(5000, t -> {
```

Comment:
We should at least make this time configgable, because really we've no idea what the ideal number should be in practice. 

But I wonder if a different algorithm might work. All we know is that the znode changed, but that change could be because the assigned brokers changed (for any partition) or because the number of partitions changed. But we know what values the configmap currently has for both those numbers. So we could:

1. poll for the topic metadata until we observe a change in either of them. 
2. If it turns out that it was a RF change then we do nothing. 
3. If it turns out the partition number changed then we proceed to reconcile. 

This would be the analogous algorithm to what we do during topic creation. Obviously there's a limit to how long you should continue to poll (I used `BackOff` for that to topic creation). This algorithm means you can potentially stand to wait for longer overall, but without introducing lots of latency, so choosing the right timeout is less critical.

Just an idea, wdyt?

Reply:
Agree.